docs/loaders/README.md

```diff
@@ -0,0 +1,77 @@
+
+# Loaders documentation
+
+Loaders in spetlr:
+
+- [SCD2Loader](#SCD2Loader)
+ 
+
+# SCD2Loader
+
+The SCD2Loader is a class designed for handling Slowly Changing Dimensions (SCD) Type 2 in PySpark dataframes. 
+This loader is particularly useful in scenarios where data evolves over time, and there is a need to maintain 
+historical versions of data records. It enables upserts to SCD2 tables, meaning it can both update existing records and
+insert new records as needed.
+
+The loader adds `ValidFrom`, `ValidTo`, `IsCurrent`, and `HashValue` columns to the source schema. The ValidFrom and ValidTo 
+columns help in tracking the duration for which a record is valid. The IsCurrent flag indicates whether a record is the 
+latest version. The HashKey is a unique identifier for each record, typically generated by hashing one or more columns of the data.
+
+Recommendations: Add [liquid clustering](https://docs.databricks.com/en/delta/clustering.html) to your source and sink table. 
+Cluster by the `join_columns` and the `HashValue` column.
+
+## Example
+Here is an example of how the SCD2Loader can be used in a PySpark application:
+
+First, consider a PySpark dataframe with the following schema and data:
+
+```markdown
+df.show()
++-------+------+----------+-------------------+
```

Comment:
As discussed earlier, PKs have a unique constraint. Maybe update examples to reflect this.